gui/default/syncthing/core/syncthingController.js

```diff
@@ -395,11 +395,80 @@ angular.module('syncthing.core')
                 && guiCfg.authMode !== 'ldap'
                 && !guiCfg.insecureAdminAccess;
 
+            // If no authentication configured, give info to user,
+            // in case of conflict we have time difference between openNoAuth and NoAuth.
+            $scope.NoAuth = initAuthorization(guiCfg);
+
             if (!hasConfig) {
                 $scope.$emit('ConfigLoaded');
             }
         }
 
+        function initAuthorization(guiCfg) {
+            var userPwd = !guiCfg.user || !guiCfg.password;
+
+            if (userPwd === true){
```

Comment:
There is logic oddness here. If we don't have `initNotification` set, we set it to one. Then, if it's set to one (which it is), we set it to zero and return `userPwd` (which we've already confirmed is `true`). I think something else is intended?

I would expect a local storage variable `authenticationWarningDismissed` or something like that. If it's set we don't show the warning. When the user dismisses the dialog we set it to one. Nothing more needed, really, I think.